pkg/controller/node/node_controller_test.go

```diff
@@ -904,6 +942,46 @@ func TestShouldDoNothing(t *testing.T) {
 	f.run(getKey(mcp, t))
 }
 
+func TestControlPlaneTopology(t *testing.T) {
```

Comment:
Sorry, I'm a bit confused by what this is trying to test. Is it checking to see if the controllerconfig topology propagates to node annotations? Where is that being checked?

Reply:
Here, we are currently checking that with a valid controlPlaneToplogy i.e. `SingleReplica` ,  setClusterConfigAnnotation() gets called and works as expected. Since, in this unit test we have created controllerConfig with `SingleReplica` and we have set node annotation `machineconfiguration.openshift.io/controlPlaneTopology` to `SingleReplica` , expected result is no node change action such as `patch`.

Later on, I am thinking of extending the test to also perform updating node annotation update action. Adding this is a bit tricky.

Reply:
Hmm, sorry for the dumb question, but I'm still not sure how we are testing that. If an action change happens as a bug, which line of code would return the error? I see we do the same for the above `TestShouldDoNothing` but I don't see any expectations for "nothing"

Reply:
when in test run() calls runController(), node-controller syncHandler(pool) gets called at https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/node/node_controller_test.go#L114 . As you know when this runs, various actions can be generated based on what all operation on node has been performed. In test we filter out list and watch actionhttps://github.com/openshift/machine-config-operator/blob/master/pkg/controller/node/node_controller_test.go#L121as these actions doesn't really update anything. For other actions like patch, additional action will be check in later line during checkAction()